pkg/controller/controller.go

```diff
@@ -136,11 +139,12 @@ func checkNamespace(kcs kubernetes.Interface, conf *config.Config) error {
 	return err
 }
 
-func getSelfDeploy(kcs kubernetes.Interface, conf *config.Config) (*appsv1.Deployment, error) {
+func getSelfDeploy(kcs kubernetes.Interface, conf *config.Config, log logr.Logger) (*appsv1.Deployment, error) {
 	deploy, err := kcs.AppsV1().Deployments(conf.NS).Get(context.Background(), conf.OperatorDeployment, metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		// It's okay if we don't find the deployment - just skip setting ownership references
-		err = nil
+		// It's okay if we don't find the deployment - just skip setting ownership references latter
+		log.Info("self deploy not found")
```

Comment:
💡 error handling suggests- just some thoughts - by doing this we are initiating a code path where there is no error.

can we still return an error with the “self deploy not dogune message”

tagt way we don’t need special case handle things.

Reply:
we aren't changing the code path in terms of the error. since this originally set err = nil we never returned an error if the self deploy wasn't found.

we could return an error but that doesn't accurately represent this function in the context of web app routing. It's okay if the self deployment isn't found and the rest of web app routing handles that case with a nil check on the deploy object.

Wrapping the deployment struct with information on whether it was found would be the alternative solution but that's overkill when `nil` is considered "not existing" in go.

Reply:
Sure, I think I will let you decide, either we should custom error or enum it for more easy understanding of this scenario. 

The ripple down effect for special cases in long run as interesting. Here we are logging a piece of information which feels like should be more upfront for user.

**but** again if don’t know the scenario so mentioned above because it kind of grocery form the way son code path are working. 

Usually using a nil return is fine but be careful of - Your code will repeat all these nil check every times you need to access pmethod of an instance, contaminating your code with many condition that have nothing to do with your feature.